doc/pages/Manual.md

```diff
@@ -7,33 +7,228 @@ primarily of use for packagers, package maintainers and repository maintainers.
+- `~/.opam/<switch>` for plain switches
+- `<switch>/_opam` for local switches, when `<switch>` is a path
+
+Switches are laid out thusly:
+- `<switch-prefix>/`: prefix of the switch, holding the installation hierarchy
+  in the UNIX `/usr` standard (with subdirectories `bin`, `lib`, `share`, `man`,
+  `doc`, `etc`...)
+- `<switch-prefix>/.opam-switch/`: holds all opam data regarding this switch
+- [`<switch-prefix>/.opam-switch/switch-config`](#switchconfig): switch-specific
+  configuration
+- [`<switch-prefix>/.opam-switch/switch-state`](#switchstate): stores the sets
+  of installed, base, pinned packages
+- `<switch-prefix>/.opam-switch/environment`: contains the environment variable
+  settings for this switch
+- `<switch-prefix>/.opam-switch/reinstall`: list of packages marked for
+  reinstallation (development packages where changes have been detected)
+- [`<switch-prefix>/.opam-switch/config/<pkgname>.config`](#ltpkgnamegtconfig):
+  installed package's, opam specific configuration
+- [`<switch-prefix>/.opam-switch/install/<pkgname>.install`](#ltpkgnamegtinstall):
+  `.install` files used to install the given package
+- `<switch-prefix>/.opam-switch/install/<pkgname>.changes`: file changes done by
+  the installation of the given package, as tracked by opam
+- `<switch-prefix>/.opam-switch/packages/<pkgname>.<version>/`: metadata of the
+  given package as it has been used for its installation
+- `<switch-prefix>/.opam-switch/packages.dev/<pkgname>/`: sources (archives or
+  VC clones) of the given packages, when not fetched from the repository
+  directly
+- `<switch-prefix>/.opam-switch/overlay/<pkgname>/`: custom definition for the
+  given pinned packages
+- `<switch-prefix>/.opam-switch/build/<pkgname>.<version>/`: temporary
+  directories where the packages are compiled
+- `<switch-prefix>/.opam-switch/remove/<pkgname>.<version>/`: temporary
+  directories used for calling the packages' `remove` commands, when those need
+  the source.
+- `<switch-prefix>/.opam-switch/backup`: snapshots of previous states of the
+  switch, and other backup files.
+
+#### Pinning
+
+Pinning is an operation by which a package definition can be created or altered
+locally in a switch.
+
+In its most simple form, `opam pin <package>`, `<package>` is bound to its
+current version and won't be changed on `opam upgrade` (assuming it is an
+existing package).
+
+`opam pin` can further be used to divert the source of a package, or even create
+a new package, using `opam pin <package> <URL>` ; this is also very useful to
+point to a local path containing a development or patched version of the package
+source. If the source you are pointing to contains a matching `<pkgname>.opam/`
+or `opam/` directory, a matching `<pkgname>.opam` or `opam` file, that will be
+used as the initial package metadata.
+
+Whenever an install, reinstall or upgrade command-line refers to a pinned
+package, opam first fetches its latest source. `opam update` otherwise updates
```

Comment:
On thing that needs to be clarified about opam update (in the description of repos) is whether opam update updates all the repos in the global list or only the repos used in the current switch (would personally prefer the latter)